ekundayo-ab/hello-books
# #152996280 View categories filter books

### What does this PR do?
This pull request ensures book categories are listed and filters book by any category selected from the list. Also, some cleanups and refactoring unrelated to this task were done.

### Description of Task to be completed?
- Retrieve all available categories from the database and display them on the **Admin** and **Shelf** page
- Filter the books on the respective pages above by the selected category from the category list

### How should this be manually tested?
### How should this be manually tested?
1. Pull in latest changes from remote repository to local repository
2. Fire up terminal run `npm install`
3. execute `npm run start:dev` and then `npm run build:dev` to start up application
4. Visit url supplied by the application usually **http://localhost:3000**
5. login with `ekprogs@gmail.com` and `dayo` for admin access, while for normal user access simply register a new user on the application
6. After logging in to the application, visit the **Admin** Dashboard create a category, create a book
7. Click on each category of the category list by the left to see the the books filtered
8. Visit the **Shelf** page do as above

### Any background context you want to provide?
Categories should be added to the database before adding a book that uses such category

### What are the relevant pivotal tracker stories?
#152996280/view categories filter books

### Screenshots (if appropriate)
N/A

### Questions:
N/A
